core/matching/query_matching.go

```diff
@@ -59,3 +60,15 @@ func QueryMatching(requestMatcher models.RequestMatcher, toMatch map[string][]st
 		Score:   score,
 	}
 }
+
+func getStringToBePassedForMatching(strArr []string) string {
```

Comment:
I think converting query params or header values to json array back and forth is a bit overkill, it won't be as efficient as strings with a delimiter. It might not be worth making this change for a very specific use case, also the JSON array may contains JSON objects instead of simple string values. I feel like user using JSON path matchers would be better off chaining JSON paths to process the JSON data rather than mixing with the Array matcher.

Reply:
btw you can take a look at this one for bumping minor version of the simulation: https://github.com/SpectoLabs/hoverfly/commit/d6c2ce4ca2e30a62545cb371ee80292a9bf1506d, we also need to update the simulation schema file: https://github.com/SpectoLabs/hoverfly/blob/master/schema.json

Reply:
> I think converting query params or header values to JSON array back and forth is a bit overkill, it won't be as efficient as strings with a delimiter. If the change is risky or has an adverse effect on wider hoverfly usages, I don't think it's worth making this change for a very specific use case.
> 
> In a real scenario, JSON array may contain JSON objects instead of just simple string values, so Array matcher might not be the right one to use anyway. If a user requires JSON path matching, they would be better off chaining the jsonPath matchers to process the JSON data rather than mixing with the Array matcher in my opinion.

@tommysitu yea completely makes sense. It will be an overkill and may have adverse affect as well so better if I wont do it.

Reply:
> btw you can take a look at this one for bumping minor version of the simulation: [d6c2ce4](https://github.com/SpectoLabs/hoverfly/commit/d6c2ce4ca2e30a62545cb371ee80292a9bf1506d), we also need to update the simulation schema file: https://github.com/SpectoLabs/hoverfly/blob/master/schema.json

@tommysitu it will be better if you can help me with this change. I don't want to mess things up while doing this.

Reply:
> btw you can take a look at this one for bumping minor version of the simulation: [d6c2ce4](https://github.com/SpectoLabs/hoverfly/commit/d6c2ce4ca2e30a62545cb371ee80292a9bf1506d), we also need to update the simulation schema file: https://github.com/SpectoLabs/hoverfly/blob/master/schema.json

I will make this change and and get back with new PR.